### PR TITLE
AVX-64: Add HTTPS support

### PR DESCRIPTION
**Pull request by dat-tran-matternet**

### CODE REVIEW

File: ntripclient.c [hunk 1]
```diff
@@ -566,6 +569,32 @@ static int encode(char *buf, int size, const char *user, const char *pwd)
...
+    size_t bytes_recv = 0;
+    CURLcode result = CURLE_OK;
+
+    do
+    {
+        result = curl_easy_recv(curl, buf, len, &bytes_recv);
+
+        // Wait for 100ms before trying again.
+        if (result == CURLE_AGAIN)
+            usleep(100 * 1000);
```
Comment (bhiln) [inline ntripclient.c]: consider defining 1ms to remove magic number:
```
unsigned int one_ms = 1000;
...
usleep(100 * one_ms);
```

File: ntripclient.c [hunk 2]
```diff
@@ -1432,8 +1461,29 @@ int main(int argc, char **argv)
...
+          char url[1024];
+          const char *proto = (args.mode == HTTPS) ? "https://" : "http://";
+          snprintf(url, sizeof(url), "%s%s", proto, server);
+          url[sizeof(url)-1] = 0;
+
+          // Set CURL options
+          curl_easy_setopt(curl, CURLOPT_URL, url);
+          curl_easy_setopt(curl, CURLOPT_PORT, atoi(port));
+          curl_easy_setopt(curl, CURLOPT_CONNECT_ONLY, 1L);
+          //curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);
```
Comment (bhiln) [inline ntripclient.c]: Please remove this line if it is no longer used

**Review by bhiln — APPROVED** [review]
Left minor suggestions, but looks great!

File: ntripclient.c (hunk 2 above)
Comment (dat-tran-matternet) [reply ntripclient.c]: Done. This line is needed time to time to debug the connection during development. I changed the setting to 0. Any developer can change back to 1 for debugging in the future.

File: ntripclient.c (hunk 1 above)
Comment (dat-tran-matternet) [reply ntripclient.c]: Done. Made a macro for SLEEP_MS.

**Review by rabad-matternet — APPROVED** [review]
This looks good but please test before merging.

**Review by AydenChubbic — APPROVED** [review]

File: ntripclient.c [hunk 3]
```diff
@@ -845,14 +845,15 @@ int main(int argc, char **argv)
       // this block of code is very long and difficult to read.  I'll attempt
       // to summarize here.
       //
-      // Lines 747 - 1080 handle UDP mode
-      // Lines 1082 - 1433 handle RTSP mode
-      // Lines 1435 - 1822 handle all the other modes
-      //
```
Comment (dat-tran-matternet) [inline ntripclient.c:736]: @rabad-matternet I removed the line numbers so we don't have to keep maintaining them. I changed it to say there are three major sections to look out for.

Comment (rabad-matternet) [reply ntripclient.c:736]: I agree that line numbers are problematic to maintain.  How about putting a comment that can be searched at each section and then state what that searchable comment is here?  

The motivation for the line numbers (and now a searchable comment) is that this function is to big to see what's going on.  So having a way to easily navigate would be nice.

Comment (dat-tran-matternet) [reply ntripclient.c:736]: Done. Great suggestion.

**Comment by dat-tran-matternet** [comment]
Will request to pull into dev instead of master.

**Pull request closed without merging**